lib/linear_support.js

```diff
@@ -9,13 +9,44 @@ var Linear = require('sylvester');
  *
  * Made available via the MIT License, text included in LICENSE.
  *
+ */
+
+/*
+ * helper method to reduce duplication for validating data types
+ *
+ * obj - The object we want to test the type of
+ * type - The expected type
+ * msg - Additional message to include on error
+ */
+var isObjectAt = function(obj, type, msg) {
+  var errorMessage = "Argument needs to be of type " + type + ".  " + msg;
+
+  switch (type) {
+    case 'Vector':
```

Comment:
I'm not a huge fan of magic strings which can be mistyped and result in a runtime error. Perhaps replace this with a static variable?